base/uk.ac.stfc.isis.ibex.ui.beamstatus/src/uk/ac/stfc/isis/ibex/ui/beamstatus/views/SynchrotronPanel.java

```diff
@@ -69,14 +72,39 @@ private Label createRow(String name) {
         display.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 1, 1));
         return display;
     }
+    
+
+   
 
     private void bind(SynchrotronObservables sync) {
         DataBindingContext bindingContext = new DataBindingContext();
-        bindingContext.bindValue(WidgetProperties.text().observe(beamCurrent), BeanProperties.value("value").observe(sync.beamCurrent));
-        bindingContext.bindValue(WidgetProperties.text().observe(beamFrequency), BeanProperties.value("value").observe(sync.beamFrequency));
-        bindingContext.bindValue(WidgetProperties.text().observe(beamEnergy), BeanProperties.value("value").observe(sync.beamEnergy));
-        bindingContext.bindValue(WidgetProperties.text().observe(bunchLength), BeanProperties.value("value").observe(sync.bunchLength));
-        bindingContext.bindValue(WidgetProperties.text().observe(extractDelay), BeanProperties.value("value").observe(sync.extractDelay));
+        
+        BeamInfoMenu beamCurrentMenu = new BeamInfoMenu(sync.beamCurrent);
+    	beamCurrent.setMenu(beamCurrentMenu.createContextMenu(this));
+        bindingContext.bindValue(WidgetProperties.text().observe(beamCurrent), BeanProperties.value("value").observe(sync.beamCurrent.updatedValue));
```

Comment:
There's a lot of repeated code here (and in `TargetStationOnePanel` and `TargetStationTwoPanel`). Since we are applying the same operation to each of the FacilityPV / Label pairs, perhaps we could pull the above lines out into its own method, something like `bindAndAddMenu` which you then just call with the observable and the label you want to bind to it